### PR TITLE
Issue #8: Request Controller

### PR DESCRIPTION
**Pull request by jtingling**
### What this PR does (required):
- Create, read, update operations on Request model for dog sitting.

### Any information needed to test this feature (required):
- Register two users.
- Set a environment variable `TOKEN` with a encoded JWT token of the dog owner;
- Re-assign `userId` and `sitterId` values to the `id` values of the  users you registered in `models.spec.js`.
- On the command line, run `npm run testRequets`.

### CODE REVIEW

File: server/controllers/request.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const Request = require('../models/Request');
+const asyncHandler = require("express-async-handler");
+
+exports.getRequests = asyncHandler(async (req, res ) => {
+    const userId = req.query.userId;
```
Comment (ezeilo-su) [inline server/controllers/request.js:5]: You need to return a `400` if this field is missing

File: server/controllers/request.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        if (!requests) {
+            res.status(404);
+            throw new Error("No request for user exists.");
+        }
+        res.status(200).json({requests});
+    });
+});
+
+exports.createRequest = asyncHandler( (req, res ) => {
+    const { userId, sitterId, date, start, end } = req.body;
```
Comment (ezeilo-su) [inline server/controllers/request.js:20]: Please validate the request body and returna `400` if any required filed is missing

File: server/controllers/request.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    .exec(function(err, requests) {
+        if (err) {
+            res.status(500);
+            throw new Error('Problem finding document.')};
+        if (!requests) {
+            res.status(404);
+            throw new Error("No request for user exists.");
+        }
+        res.status(200).json({requests});
+    });
```
Comment (ezeilo-su) [inline server/controllers/request.js:16]: You can use `await` here since the function is an `async` type

File: server/controllers/request.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        date,
+        start,
+        end,
+    }, function(err, _) {
+        if (err) {
+            res.status(500);
+            throw new Error("Problem creating document.");
+        };
+        res.status(201).json({ message: "Request created."});
+    });
```
Comment (ezeilo-su) [inline server/controllers/request.js:33]: Use `await` with an `async` function

File: server/controllers/request.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    const request = await Request.findOne({userId, sitterId, date, start, end}).exec();
+    if (!request) {
+        res.status(404);
+        throw new Error('Request not found.');
+    }
+    request.accepted = accepted;
+    request.declined = declined;
+    await request.save().then( savedRequest => {
+        res.status(200).json({ savedRequest });
+    });
```
Comment (ezeilo-su) [inline server/controllers/request.js:48]: Same as the comments above

File: server/models/Request.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+const mongoose = require("mongoose");
+
+const requestSchema = new mongoose.Schema({
+    userId: {
+        type: mongoose.Schema.Types.ObjectId,
+        required: true,
```
Comment (ezeilo-su) [inline server/models/Request.js:6]: This should reference the `User` model

File: server/models/Request.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    sitterId: {
+        type: mongoose.Schema.Types.ObjectId,
+        required: true,
```
Comment (ezeilo-su) [inline server/models/Request.js:10]: Add a `ref` to the `User` model

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Good work with this one. A few comments to address, though.

**Review by ezeilo-su — COMMENTED** [review]
Thank you for making the requested changes. Please wait for the team to approve before you can merge.
Happy coding!

**Review by moffatethan — APPROVED** [review]

**Pull request merged**